### PR TITLE
feat: Add microagent knowledge integration to resolver workflow

### LINKED ISSUE #6191: Feat: Microagents Support for Resolver
**What problem or use case are you trying to solve?**
Perhaps I am mistaken, but it appears that neither `.openhands/microagents/repo.md` nor microagent markdowns in `.openhands/microagents/knowledge/` nor `.openhands/microagents/tasks/` are recognized and used by the Resolver.

**Describe the UX of the solution you'd like**
The Resolver should:
1. Automatically detect and load microagents from `.openhands/microagents/` and its subdirectories (`repo.md`, `knowledge/`, and `tasks/`) without manual configuration, which may be several steps into the Resolver's thinking process.
2. Trigger knowledge agents contextually based on keywords detected in conversations or file content.
3. Emit directly or indirectly in the action output logs microagents.
4. Display errors or warnings if microagents have invalid syntax or fail to load, ensuring users can troubleshoot issues.

**Do you have thoughts on the technical implementation?**
1. **Loading Microagents**:
   - Implement a directory scanning feature in the Resolver to locate and parse files in `.openhands/microagents/`.
   - Use YAML parsers to validate and load metadata from the frontmatter of each microagent file.
   - Prioritize loading order:
     1. `repo.md` for repository-specific settings.
     2. `knowledge/` for context-based knowledge triggers.
     3. `tasks/` for interactive workflows.
   
2. **Integration with OpenHands**:
   - Enhance the Resolver logic to allow dynamic activation of microagents based on triggers (e.g., keywords).
   - Treat the thinking Resolver agent as the user for task agents
   - Maintain a cache of loaded microagents for quick access and reduce redundant parsing.

3. **Testing**:
   - Write unit tests for the Resolver to cover edge cases like missing directories or invalid microagent structures.
   - Create integration tests to ensure microagents interact correctly within OpenHands workflows.

**Describe alternatives you've considered**
1. **Manual Configuration**: Duplicate the frontmatter from all microagents into .openhands_instructions. This approach is less user-friendly and increases the risk of misconfiguration.

### PR DESCRIPTION
**Pull request by h55nick**
Fixes All-Hands-AI/OpenHands#6191

This PR adds integration between the resolver workflow and microagent knowledge system.

### Changes
- Add microagent trigger checking in resolver's `get_context_from_external_issues_references`
- Include relevant knowledge based on issue/PR content and comments
- Add tests demonstrating microagent integration with examples

### Implementation Details
- Uses existing microagent trigger system
- Checks all issue content (body, comments, review threads) for triggers
- Maintains consistency with how microagents are used elsewhere
- No new microagents needed - uses existing knowledge agents

### Testing
Added tests that verify:
1. Knowledge is included when triggers match
2. Knowledge is excluded when no triggers match
3. Triggers work across issue body and comments

### Example
If an issue mentions 'flarglebargle', and we have a knowledge microagent with that trigger, its knowledge will be included in the resolver context.

### CODE REVIEW

File: tests/unit/resolver/test_microagent_integration.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+def test_resolver_uses_knowledge_microagents():
+    """Test that resolver picks up relevant knowledge microagents based on triggers."""
+    # Create a test knowledge microagent
+    test_knowledge_content = """---
+name: flarglebargle
+type: knowledge
+version: 1.0.0
+agent: CodeActAgent
+triggers: ["flarglebargle", "bargling"]
```
Comment (h55nick) [inline tests/unit/resolver/test_microagent_integration.py]: I think this should be 
```
triggers:
- flarglebargle
- bargling
```

File: tests/unit/resolver/test_microagent_integration.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+
+def test_resolver_uses_knowledge_microagents_with_comments():
+    """Test that resolver checks triggers in comments and review threads."""
+    test_knowledge_content = """---
+name: performance
+type: knowledge
+version: 1.0.0
+agent: CodeActAgent
+triggers: ["performance", "optimization"]
```
Comment (h55nick) [inline tests/unit/resolver/test_microagent_integration.py]: same format issue here.
